parity-util-mem/src/memory_stats_noop.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[derive(Clone, Debug)]
+pub struct Unimplemented;
+pub use Unimplemented as Error;
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for Unimplemented {
+	fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+		fmt.write_str("unimplemented")
+	}
+}
+
+#[derive(Clone)]
+pub struct MemoryAllocationTracker {}
+
+impl MemoryAllocationTracker {
+	pub fn new() -> Result<Self, Error> {
+		Err(Error)
+	}
+
+	pub fn snapshot(&self) -> Result<crate::MemoryAllocationSnapshot, Error> {
+		unimplemented!();
```

Comment:
What's the rationale of panicking instead of return an `Err`?
The struct is possible to construct w/o `::new` method, so this is reachable.

Reply:
It's not

Reply:
The struct's not directly exposed outside of the crate, so while it is possible to construct it *within* the crate you cannot construct it from the outside. So there's probably not much point in implementing `snapshot` if it's never going to be called.

Reply:
Hmm, seems a bit dangerous to rely on developer discipline for this. Returning an error seems just as easy, or…?